### PR TITLE
Stripe auth email sending; quick tidy up

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Tidies up some opaque code around sending Stripe authorization emails where the `StripeScaPaymentAuthorize` class was _sometimes_ being quietly overriden with the `SendAuthorizationEmails` module using `#extend` (changing it's behaviour in an obscure way, but only in some cases).

Also reduces the total number of places where we define a `return_url` when processing Stripe payments (the main point here). There's still a lot _more_ places where we define the `return_url` for Stripe; I'd like to reduce it to **one** eventually... :sweat_smile:

Soundtrack: [Tesselate](https://www.youtube.com/watch?v=Qg6BwvDcANg) :musical_note: 

#### What should we test?

~~Green build should be okay here.~~

Trigger auth emails when placing order as admin and through subscription.

#### Release notes

Merge SendAuthorizationEmails and StripeScaPaymentAuthorize

Changelog Category: Technical changes

### CODE REVIEW

File: engines/order_management/app/services/order_management/order/stripe_sca_payment_authorize.rb [hunk 1]
```diff
@@ -19,8 +20,15 @@ def call!(redirect_url = full_order_path(@order))
           @order.errors.add(:base, I18n.t('authorization_failure'))
         end
 
-        @payment
+        return @payment unless @payment.requires_authorization? && off_session
+
+        PaymentMailer.authorize_payment(@payment).deliver_now
+        PaymentMailer.authorization_required(@payment).deliver_now
```
Comment (achauve) [inline engines/order_management/app/services/order_management/order/stripe_sca_payment_authorize.rb]: It's really a cosmetic detail and maybe not the ruby way either, but as a reader I would have found it clearer with somthing like that:

```ruby
need_asynchronous_payment_authorization = @payment.requires_authorization? && off_session

if need_asynchronous_payment_authorization
    PaymentMailer.authorize_payment(@payment).deliver_now
    PaymentMailer.authorization_required(@payment).deliver_now
else
    return @payment
end

**Review by achauve — APPROVED** [review]

File: engines/order_management/app/services/order_management/order/stripe_sca_payment_authorize.rb (hunk 1 above)
Comment (achauve) [reply engines/order_management/app/services/order_management/order/stripe_sca_payment_authorize.rb]: Just saw afterwards that you also edited the code so my comment is not very useful.

**Comment by Matt-Yorkley** [comment]
Ah I missed a detail here; there are two emails which can be sent when authorization is required, but one of them (to notify the hub manager) only gets sent for subscriptions and not in other cases.

I'll add a commit...

**Review by mkllnk — APPROVED** [review]
Great work!

I know that this is a bit of hassle to test and there is test coverage but I would feel more comfortable if we did a test on staging.

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley ,

All good here. In both tested scenarios, the email requesting for authorization was received, among the others:

- Placed subscription order, requesting a Stripe payment, and received all the (6) expected emails.
- Placed a BO order, requesting a Stripe payment and received the (2) expected emails.

Merging!

PS: spotted this issue in the `/admin/order_cycles` page - #8665

**Pull request merged**